### PR TITLE
API: Initial `finch_backend`

### PR DESCRIPTION
**Pull request by mtsokol**
Initial `finch_backend` setup.

### CODE REVIEW

File: ci/environment.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ dependencies:
   - pytest
   - pytest-cov
   - pre-commit
-  - julia
+  - julia=1.10.1
```
Comment (hameerabbasi) [inline ci/environment.yml:11]: Should be `1.10.0`, since `1.10.1` is what broke.

Comment (mtsokol) [reply ci/environment.yml]: Ah, sure!

File: ci/environment.yml [hunk 2]
```diff
@@ -1,5 +1,14 @@
 name: sparse-dev
 channels:
   - conda-forge
+  - nodefaults
 dependencies:
-  - pip
```
Comment (hameerabbasi) [inline ci/environment.yml:5]: Let's keep `pip`, as we use it to install `sparse`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -34,6 +34,9 @@ jobs:
           key:
             test-${{ matrix.os }}-conda-py${{ matrix.python }}-${{ env.CACHE_NUMBER }}-${{
             hashFiles('ci/environment.yml') }}
+      - uses: julia-actions/setup-julia@v1.9
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:37]:
```suggestion
      - uses: julia-actions/setup-julia@v1
```

If it exists. 😉

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Couple of small comments.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -46,7 +49,7 @@ jobs:
           pip install -e .[tests]
       - name: Run tests
         run: |
-          pytest --pyargs sparse
+          pytest --pyargs sparse/tests/test_backends.py
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:52]:
```suggestion
          pytest --pyargs sparse
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -34,6 +38,9 @@ jobs:
           key:
             test-${{ matrix.os }}-conda-py${{ matrix.python }}-${{ env.CACHE_NUMBER }}-${{
             hashFiles('ci/environment.yml') }}
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.julia_version }}
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:43]:
```suggestion
          version: 1.10.0
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -7,14 +7,18 @@ jobs:
...
         include:
           - os: macos-latest
             python: '3.9'
+            julia_version: '1.10.2'
           - os: windows-latest
             python: '3.9'
+            julia_version: '1.10.2'
           - os: ubuntu-latest
             python: '3.9'
+            julia_version: '1.10.0'
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:21]: Let's undo these changes.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

**Comment by mtsokol** [comment]
@hameerabbasi The PR is ready from my side. It contains initial `finch-tensor` setup where we can run and test existing `Tensor` API. In the next steps I will start moving `pydata_backend` tests to a common `tests` directory and test against common Array API.

File: sparse/finch_backend/__init__.py [hunk 7]
```diff
@@ -1,4 +1,11 @@
-Tensor = ()
+try:
+    import finch  # noqa: F401
+except ModuleNotFoundError:
+    raise ImportError("Finch not installed. Run `pip install sparse[finch]` to enable Finch backend") from None
```
Comment (hameerabbasi) [inline sparse/finch_backend/__init__.py:4]:
```suggestion
except ModuleNotFoundError as e:
    raise ImportError("Finch not installed. Run `pip install sparse[finch]` to enable Finch backend") from e
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
One minor change and this looks good to go.

**Review by hameerabbasi — APPROVED** [review]
